### PR TITLE
helium-wallet 1.6.6 (new formula)

### PR DESCRIPTION
**Pull request by bddvlpr**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

New formula publish request for [helium-wallet-rs](https://github.com/helium/helium-wallet-rs) v1.6.6, revision `4becd5fd9f817ecfee00dba3177a64c7fc223bd1`.

### CODE REVIEW

File: Formula/helium-wallet.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  license "Apache-2.0"
+  head "https://github.com/helium/helium-wallet-rs.git", { using: :git, branch: "master" }
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "cmake" => :build
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/helium-wallet.rb:14]:
```suggestion
  depends_on "rust" => :build
  uses_from_macos "llvm"
```

File: Formula/helium-wallet.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  def install
+    system "cargo", "install", *std_cargo_args
+    prefix.install "README.md"
+    prefix.install "LICENSE"
```
Comment (carlocab) [inline Formula/helium-wallet.rb:19]:
```suggestion
```
`brew` does these automatically.

File: Formula/helium-wallet.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  test do
+    # Currently disabled since `helium-wallet create basic` requires user input.
```
Comment (carlocab) [inline Formula/helium-wallet.rb:23]: Let's simulate that user input then. We really need a test that does more than `--version` or `--help`.

Comment (bddvlpr) [reply Formula/helium-wallet.rb:23]: Could you elaborate on the best way of doing this? I've seen some people using Expect but I'm unsure if this is the way to do it.

Comment (carlocab) [reply Formula/helium-wallet.rb:23]: You could try using Expect, or use Ruby's PTY module. The latter is probably simpler -- see formulae that do `require "pty"` for examples.

Comment (bddvlpr) [reply Formula/helium-wallet.rb:23]: Never worked with PTY before but for some reason it won't spawn the process. Currently using the following:
```rb
  test do
    require "pty"

    key = testpath/"test-wallet.key"
    _, w, pid = PTY.spawn(bin/"helium-wallet", "create", "basic", "-o", key)
    sleep 2
    w.write "test-key-password\r"
    sleep 2
    w.write "test-key-password\r"
    sleep 5

    assert_predicate key, :exist?
  end
```

**Pull request closed without merging**